damus/Models/HomeModel.swift

```diff
@@ -487,7 +489,40 @@ class HomeModel: ObservableObject {
             return
         }
         
-        handle_last_event(ev: ev, timeline: .notifications)
+        if handle_last_event(ev: ev, timeline: .notifications),
+           damus_state.contacts.follow_state(ev.pubkey) == .follows,
+            let type = ev.known_kind {
+
+            if type == .text,
+               damus_state.settings.mention_notification {
+                for block in ev.blocks(damus_state.keypair.privkey) {
+                    if case .mention(let mention) = block, mention.ref.ref_id == damus_state.keypair.pubkey,
+                       let displayName = damus_state.profiles.lookup(id: ev.pubkey)?.display_name {
+                        let justContent = NSAttributedString(render_note_content(ev: ev, profiles: damus_state.profiles, privkey: damus_state.keypair.privkey).content).string
+                        create_notification(displayName: displayName, conversation: justContent, type: type)
+                    }
+                }
+            } else if type == .boost,
+                      damus_state.settings.repost_notification,
+                      let displayName = damus_state.profiles.lookup(id: ev.pubkey)?.display_name {
+
+                let justContent = NSAttributedString(render_note_content(ev: ev, profiles: damus_state.profiles, privkey: damus_state.keypair.privkey).content).string
+                if let jsonData = justContent.data(using: .utf8),
+                   let jsonDict = try? JSONSerialization.jsonObject(with: jsonData, options: []) as? [String: Any],
+                   let content = jsonDict["content"] as? String {
```

Comment:
I'm confused as to the point of this? it seems to be rendering the note content, serializing as json and then pulling out the content again? huh?

Reply:
ok nm I see now, but we have a better way of doing this. will fix.

Reply:
@jb55 -
specially for event with boost kind, when I fetch the note content using render_note_content() it will give me a json data like below, and i have to manually convert into string (however, this is not the case from .text kind)
<img width="564" alt="Screen Shot 2023-04-03 at 7 43 26 PM" src="https://user-images.githubusercontent.com/120697811/229650430-7299b931-47b2-4a02-8230-31b27df028d3.png">
<img width="590" alt="Screen Shot 2023-04-03 at 7 43 37 PM" src="https://user-images.githubusercontent.com/120697811/229650433-b13b8efb-cf52-4c64-9836-cd4f9b89cd3b.png">